### PR TITLE
feat: adds ability to upload audit-report to s3

### PR DESCRIPTION
**Pull request by as1729**
## Description
- Adds a file to s3
- Generates a pre-signed URL for the uploaded file

## Screenshots / Demo Video
https://www.loom.com/share/c46d7230c7de4e1295abcef4663bc2e7

### Automated and Unit Tests
- [x] Added Unit tests

### Manual tests for Reviewer
- [ ] Added steps to test feature/functionality manually

## Checklist
- [x] Provided ticket and description
- [x] Provided screenshots/demo
- [x] Provided testing information
- [x] Provided adequate test coverage for all new code
- [x] Added PR reviewers

### CODE REVIEW

**Review by TylerHendrickson — CHANGES_REQUESTED** [review]

File: packages/server/src/arpa_reporter/lib/audit-report.js
Comment (pearkes) [inline packages/server/src/arpa_reporter/lib/audit-report.js:174]: Is it possible to configure the S3 client to point at localstack without this conditional, purely with environment variables? It seems a bit safer to have the uploader be totally unaware of the difference between localstack S3 and real S3 and have the configuration be purely driven via the environment, eg. test or dev and `.env`.

**Review by pearkes — COMMENTED** [review]
My comment is more general, and may be something to address in #1161, the actual testing looks like a great improvement here!

File: packages/server/src/arpa_reporter/lib/audit-report.js
Comment (TylerHendrickson) [reply packages/server/src/arpa_reporter/lib/audit-report.js:174]: Unfortunately (and frustratingly), [no, at least not natively (yet)](https://github.com/aws/aws-sdk/issues/229). However, a more general pattern could be to create a centralized AWS SDK config object to reuse when instantiating any service client. It wouldn't obviate the need for the conditional check, but it would provide consistent behavior divorced from any specific use-case. We could also choose a more generic environment variable name like `$AWS_ENDPOINT_URL` to make it less specific to LocalStack and something that wouldn't need to be wrapped in a conditional. But no AWS SDK for any language seems to provide an option to configure this automatically.

This seems like something worth exploring in (or as a follow-up to) #653 (cc @jakekreider).

File: packages/server/src/arpa_reporter/lib/audit-report.js
Comment (as1729) [reply packages/server/src/arpa_reporter/lib/audit-report.js]: Done 👍

File: packages/server/src/arpa_reporter/lib/audit-report.js
Comment (as1729) [reply packages/server/src/arpa_reporter/lib/audit-report.js]: Done 👍

File: packages/server/src/arpa_reporter/lib/audit-report.js
Comment (as1729) [reply packages/server/src/arpa_reporter/lib/audit-report.js]: Done 👍

File: packages/server/src/arpa_reporter/lib/audit-report.js
Comment (TylerHendrickson) [inline packages/server/src/arpa_reporter/lib/audit-report.js:184]: The "localstack" hostname here should be replaced with `${process.env.LOCALSTACK_HOSTNAME}` as well. Additionally, `$EDGE_PORT` might not always be set, so we _should_ provide a fallback value of `4566` when `process.env.EDGE_PORT` is null. 

```suggestion
        const endpoint = new AWS.Endpoint(`http://${process.env.LOCALSTACK_HOSTNAME}:${process.env.EDGE_PORT | 4566}`);
```

File: packages/server/src/arpa_reporter/lib/audit-report.js
Comment (TylerHendrickson) [inline packages/server/src/arpa_reporter/lib/audit-report.js:202]: Were you still planning on updating this to send an API URL with a new redirect endpoint, or holding off on that for now?

**Review by TylerHendrickson — CHANGES_REQUESTED** [review]
One more small change requested, as well as a question.

File: packages/server/src/arpa_reporter/lib/audit-report.js
Comment (as1729) [reply packages/server/src/arpa_reporter/lib/audit-report.js]: Done 👍  I made those changes but got lost when I committed incorrectly. Should be fixed now.

File: packages/server/src/arpa_reporter/lib/audit-report.js
Comment (as1729) [reply packages/server/src/arpa_reporter/lib/audit-report.js:202]: I'm working on it in a separate branch. It needed a bit more refactoring so its a taking a bit to change up the tests. I'll have that PR ready this afternoon, but wanted to merge this so we can see if there's additional terraform changes needed in case the general flow is failing in staging.

**Review by TylerHendrickson — APPROVED** [review]

**Pull request merged**
